app/src/main/java/com/codepath/apps/critterfinder/services/PetSearch.java

```diff
@@ -130,5 +138,57 @@ public PetModel getNextPet() {
             // TO DO implement another search call
         }
     }
+
+    /**
+     * Fetch a list of breeds for a particular animal
+     * @param species
+     * @param breedsReceivedCallback
+     */
+    public void fetchBreeds(final SearchFilter.Species species, final PetBreedsReceivedCallback breedsReceivedCallback) {
+        client.getBreedList(species, new TextHttpResponseHandler() {
+            @Override
+            public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
+                breedsReceivedCallback.onBreedsFailed(species);
+            }
+
+            @Override
+            public void onSuccess(int statusCode, Header[] headers, String responseString) {
+                List<Breed> breeds = null;
+                if (responseString != null) {
+                    breeds = parseBreedsFromJSON(responseString);
+                }
+                breedsReceivedCallback.onBreedsReceived(species, breeds);
+            }
+        });
+    }
+
+    // Private helper class to allow gson to extract breeds out of a deeply nested response
+    private class BreedResponse {
+        PetFinder petfinder;
+        private class PetFinder {
+            Breeds breeds;
+            private class Breeds {
+                List<Breed> breed;
+            }
+        }
+        public List<Breed> getBreeds() { return petfinder.breeds.breed; }
+    }
+
+    private List<Breed> parseBreedsFromJSON(String response) {
+        Gson gson = new GsonBuilder().
+                setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).
+                excludeFieldsWithModifiers(Modifier.FINAL, Modifier.TRANSIENT, Modifier.STATIC).
+                create();
+        BreedResponse breedResponse = gson.fromJson(response, BreedResponse.class);
+        return breedResponse.getBreeds();
```

Comment:
I get lazy about manually parsing JSON. I used the GSON framework to create an array of Breed models from the pet finder response for me.